lib/typescript/render/index.tsx

```diff
@@ -1,26 +1,19 @@
 import React from 'react';
-import {Message, Conversation} from 'httpclient';
-import TextRender from './messages/TextRender/index';
+import {renderProviders} from './renderProviders';
+import {getSource} from 'httpclient';
 
-type RenderLibraryProps = {
-  message: Message;
-  source: string;
-  currentConversation?: Conversation;
-  prevWasContact?: boolean;
-  isContact: boolean;
-  nextIsSameUser?: boolean;
-};
+import {Text} from './components/Text';
+import {getSharedComponentProps, MessageRenderProps} from './shared';
 
-const RenderLibrary = (props: RenderLibraryProps) => {
-  const {message, currentConversation, prevWasContact, isContact, nextIsSameUser} = props;
-  return (
-    <TextRender
-      conversation={currentConversation}
-      message={message}
-      showAvatar={!prevWasContact && isContact}
-      showSentAt={!nextIsSameUser}
-    />
-  );
-};
+export const SourceMessage = (props: MessageRenderProps) => {
+  const source = getSource(props.conversation);
 
-export default RenderLibrary;
+  const provider = renderProviders[source];
+
+  try {
+    return provider(props);
+  } catch (e) {
+    console.error(e);
+    return <Text {...getSharedComponentProps(props)} text="Could not render this content" />;
```

Comment:
For resilience reason we by default assume that we cannot render content. After all, we don't control the content schema so the render library is a "best effort" approach to solving the problem.